internal/pkg/aws/s3/s3.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package s3 contains utility functions for Amazon Simple Storage Service Client.
+package s3
+
+import (
+	"fmt"
+	"io"
+	"path"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/google/uuid"
+)
+
+const (
+	artifactDirName = "manual"
+)
+
+type s3Client interface {
+	Upload(input *s3manager.UploadInput, options ...func(*s3manager.Uploader)) (*s3manager.UploadOutput, error)
+}
+
+// Service wraps an Amazon Simple Storage Service client.
+type Service struct {
+	s3Svc s3Client
+}
+
+// New returns a Service configured against the input session.
+func New(s *session.Session) *Service {
+	return &Service{
+		s3Svc: s3manager.NewUploader(s),
+	}
+}
+
+// PutArtifact uploads data to a S3 bucket under a random path that ends with the file name
+// and returns its path in the bucket.
+func (s *Service) PutArtifact(bucket, fileName string, data io.Reader) (string, error) {
+	id, err := uuid.NewRandom()
```

Comment:
It might be nice to use the epoch, instead. Just so there's some sense of ordering.